lib/syntax_tree/node.rb

```diff
@@ -6420,17 +6333,21 @@ class IfNode < Node
     # [Statements] the expressions to be executed
     attr_reader :statements
 
-    # [nil | Elsif | Else] the next clause in the chain
+    # [nil | IfNode | Else] the next clause in the chain
     attr_reader :consequent
 
     # [Array[ Comment | EmbDoc ]] the comments attached to this node
     attr_reader :comments
 
-    def initialize(predicate:, statements:, consequent:, location:)
+    # [ String ] the opening of the conditional statement
+    attr_reader :beginning
```

Comment:
Instead of attaching the value of the token, could you attach the token itself? That's consistent with some other places where we've done this. Also, would you be able to make it the first child mentioned/the first one in `initialize` and `copy`? That's just closer to YARP so it's the directly we're going anyway.